docs/guide/search/extend_elasticsearch.md

```diff
@@ -0,0 +1,420 @@
+
+Next, add the Sort Clause class itself in `src/Query/SortClause/ScoreSortClause.php`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\SortClause;
+
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+final class Score extends SortClause
+{
+    public function __construct(string $sortDirection = Query::SORT_ASC)
+    {
+        parent::__construct('_score', $sortDirection);
+    }
+}
+```
+
+Sort Clauses can be valid for both content and Location search.
+To choose the search type, use either `content` or `location` in the tag when registering the visitor as a service:
+
+``` yaml
+services:
+    App\Query\SortClause\ScoreVisitor:
+        tags:
+            - { name: ezplatform.search.elasticsearch.query.content.sort_clause_visitor }
+            - { name: ezplatform.search.elasticsearch.query.location.sort_clause_visitor }
+```
+
+## Custom Facet
+
+To create a custom search Facet for use with Elastic search, create a Facet class and a Facet builder.
+You also need to add a visitor and a result extractor.
+
+The following example shows how to create a Facet that filters results according to their Content Type group.
+
+`src/Query/ContentTypeGroupFacet`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\Facet;
+
+use eZ\Publish\API\Repository\Values\Content\Search\Facet;
+
+/**
+ * This class holds counts of content with content type.
+ */
+final class ContentTypeGroupFacet extends Facet
+{
+    /**
+     * An array with ContentTypeGroup::$id as key and count of matching content objects as value.
+     *
+     * @var int[]
+     */
+    public $entries = [];
+}
+```
+
+`src/Query/ContentTypeGroupFacetBuilder`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\FacetBuilder;
+
+use eZ\Publish\API\Repository\Values\Content\Query\FacetBuilder;
+
+final class ContentTypeGroupFacetBuilder extends FacetBuilder
+{
+}
+```
+
+`src/Query/ContentTypeGroupFacetBuilderVisitor`:
+
+``` php
+<?php
+
+declare(strict_types=1);
+
+namespace App\Query\FacetBuilder;
+
+use eZ\Publish\API\Repository\Values\Content\Query\FacetBuilder;
+use Ibexa\Platform\Contracts\ElasticSearchEngine\Query\FacetBuilderVisitor;
+use Ibexa\Platform\Contracts\ElasticSearchEngine\Query\LanguageFilter;
+
+/**
+ * Example (simplified) visitor implementation for ContentTypeGroupFacetBuilder
+ */
+final class ContentTypeGroupFacetBuilderVisitor implements FacetBuilderVisitor
+{
+    public function supports(FacetBuilder $builder, LanguageFilter $languageFilter): bool
+    {
+        return $builder instanceof ContentTypeGroupFacetBuilder;
+    }
+
+    public function visit(FacetBuilderVisitor $dispatcher, FacetBuilder $builder, LanguageFilter $languageFilter): array
+    {
+        return [
+            'terms' => [
+                'field' => 'content_type_group_id_mid',
```

Comment:
The example from https://github.com/ezsystems/ezplatform-elastic-search-engine/pull/1 uses `                'field' => 'content_type_group_id_id',`, where is the `mid` coming from?

Reply:
`content_type_group_id` is declared as `MultipleIdentifierField` so it's fine. See https://github.com/ezsystems/ezplatform-elastic-search-engine/blob/44bee6a56e28ad8cec30652822ef1d26cbee3658/src/lib/DocumentMapper/DocumentFactory.php#L425-L429